src/pocolog-extract_main.cpp

```diff
@@ -0,0 +1,196 @@
+#include "LogFile.hpp"
+#include "Index.hpp"
+#include "IndexFile.hpp"
+#include "InputDataStream.hpp"
+#include <iostream>
+#include <boost/foreach.hpp>
+#include <typelib/csvoutput.hh>
+#include <sstream>
+#include <boost/program_options.hpp>
+
+
+using namespace pocolog_cpp;
+
+/*!
+ * \brief fast_forward_to
+ * \param dataStream
+ * \param time
+ * \return return numSamples if given time if no sample older than time is in stream
+ */
+size_t fast_forward_to(InputDataStream *dataStream, const base::Time& time)
+{
+    size_t idx = 0;
+    while(true)
+    {
+        if(idx >= dataStream->getFileIndex().getNumSamples()){
+            return dataStream->getFileIndex().getNumSamples();
+        }
+
+        if(dataStream->getFileIndex().getSampleTime(idx) > time){
+            return idx;
+        }
+        idx++;
+    }
+}
+
+void extract(InputDataStream *stream,
+             const base::Time& start_time, const base::Time& stop_time)
+{
+    std::string sep = ",";
+    std::string sdelim = "\"";
+
+
+    // Fast forward to start time
+    size_t idx = 0;
+    if(!start_time.isNull()){
+        idx = fast_forward_to(stream, start_time);
+    }
+
+    size_t stop_idx = stream->getSize();
+    if(!stop_time.isNull()){
+        stop_idx = fast_forward_to(stream, stop_time);
+    }
+
+    // Write CSV header
+    std::cout << Typelib::csv_header(*stream->getType(), stream->getName(), sep, sdelim) << std::endl;
```

Comment:
would also work without https://github.com/rock-core/tools-typelib/pull/3 if `sdelim` is not passed here